tsdb/functions.go

```diff
@@ -954,6 +961,60 @@ func (p *positionOut) lessKey(i, j int) bool {
 	return false
 }
 
+// compares types and if they differ, attempts to coerce them to
+// floating point. Returns the value unchanged if it cannot be converted.
+func typeCompare(a, b interface{}) (int, interface{}, interface{}) {
+	const (
+		stringWeight = iota
+		boolWeight
+		intWeight
+		floatWeight
+	)
+
+	va := reflect.ValueOf(a)
+	vb := reflect.ValueOf(b)
+
+	vakind := va.Type().Kind()
+	vbkind := vb.Type().Kind()
+
+	// same kind. Ordering is dependent on value
+	if vakind == vbkind {
+		return 0, a, b
+	}
+	wa, a := inferFloat(va)
+	wb, b := inferFloat(vb)
+	if wa < wb {
+		return -1, a, b
+	} else if wa == wb {
+		return 0, a, b
+	}
+	return 1, a, b
+}
+
+// returns a weighting and if applicable, the value coerced to a float
+func inferFloat(v reflect.Value) (weight int, value interface{}) {
+	const (
+		stringWeight = iota
+		boolWeight
+		intWeight
+		floatWeight
+	)
+	kind := v.Kind()
+	switch kind {
+	case reflect.Uint64, reflect.Uint32, reflect.Uint16, reflect.Uint8:
+		return intWeight, float64(v.Uint())
```

Comment:
If we are using all floats, should we return floatWeight's only and do away with intWeight altogether?